src/dune_pkg/opam_repo.mli

```diff
@@ -21,6 +21,9 @@ val of_opam_repo_dir_path
   -> Path.t
   -> t
 
+(** [of_git_repo git source] loads the data through git *)
+val of_git_repo : git:Path.t -> repo_id:Repository_id.t option -> string -> t Fiber.t
```

Comment:
What's the point of the `repo_id` argument here?

Reply:
We save it into the lock dir

Reply:
> What's the point of the `repo_id` argument here? It doesn't seem to be used.

It is used, you can use it to check out a specific revision of the repository. This is useful if you want to reproduce a very specific version of the repo and run the solver again.